channeldb/graph.go

```diff
@@ -1051,6 +1052,40 @@ func (l *LightningNode) AddPubKey(key *btcec.PublicKey) {
 	copy(l.PubKeyBytes[:], key.SerializeCompressed())
 }
 
+// NodeAnnouncement retrieves the latest node announcement of the node.
+func (l *LightningNode) NodeAnnouncement(signed bool) (*lnwire.NodeAnnouncement, error) {
+	if !l.HaveNodeAnnouncement {
+		return nil, errors.New("node does not have node announcement")
+	}
+
+	alias, err := lnwire.NewNodeAlias(l.Alias)
+	if err != nil {
+		return nil, err
+	}
+
+	nodeAnn := &lnwire.NodeAnnouncement{
+		Features:  l.Features.RawFeatureVector,
+		NodeID:    l.PubKeyBytes,
+		RGBColor:  l.Color,
+		Alias:     alias,
+		Addresses: l.Addresses,
+	}
+
+	if !signed {
+		return nodeAnn, nil
+	}
+
+	sig, err := lnwire.NewSigFromRawSignature(l.AuthSigBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	nodeAnn.Signature = sig
+	nodeAnn.Timestamp = uint32(l.LastUpdate.Unix())
```

Comment:
This should be set regardless of `!signed`

Reply:
Fixed.